goagen/gen_controller/generator.go

```diff
@@ -0,0 +1,329 @@
+package gencontroller
+
+import (
+	"flag"
+	"fmt"
+	"net"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/goadesign/goa/design"
+	"github.com/goadesign/goa/goagen/codegen"
+	"github.com/goadesign/goa/goagen/utils"
+)
+
+//NewGenerator returns an initialized instance of a JavaScript Client Generator
+func NewGenerator(options ...Option) *Generator {
+	g := &Generator{}
+
+	for _, option := range options {
+		option(g)
+	}
+
+	return g
+}
+
+// Generator is the application code generator.
+type Generator struct {
+	API       *design.APIDefinition // The API definition
+	OutDir    string                // Path to output directory
+	DesignPkg string                // Path to design package, only used to mark generated files.
+	Target    string                // Name of generated "app" package
+	Force     bool                  // Whether to override existing files
+	Pkg       string                // Name of the generated package
+	Resource  string                // Name of the generated file
+	genfiles  []string              // Generated files
+}
+
+// Generate is the generator entry point called by the meta generator.
+func Generate() (files []string, err error) {
+	var (
+		outDir, designPkg, target, ver, res, pkgName string
+		force                                        bool
+	)
+
+	set := flag.NewFlagSet("controller", flag.PanicOnError)
+	set.StringVar(&outDir, "out", "", "")
+	set.StringVar(&designPkg, "design", "", "")
+	set.StringVar(&target, "pkg", "app", "")
+	set.StringVar(&pkgName, "pkg-name", "controller", "")
+	set.StringVar(&res, "res", "controller", "")
```

Comment:
The result could be main or controller or whatever... that it would be the name of the generated file (analogous to the name of the MainFile in `gen_main`).